### ISSUE TITLE
Random Graph State Sampling

### ISSUE THREAD

**Issue by jchung2020** [quantum-advantage-tracker/quantum-advantage-tracker.github.io#151]
### Name

ibm_boston_rgs_nq70_depth70_checks27_basis_fstate

### Circuit

random_graph_sampling_nq70_depth70_checks27_basis_fstate

### Value

0.37

### Method

Quantum

### Method proof

We provide a summary of the random graph state sampling problem below, a more detailed writeup is available in the [circuit instance description](https://github.com/quantum-advantage-tracker/quantum-advantage-tracker.github.io/tree/main/data/classically-verifiable-problems/circuit-models/random_graph_sampling).

## Random Graph State Sampling

We present an experiment to demonstrate quantum advantage by sampling from random graph states in a non-Clifford basis. We conjecture that this is average-case #P-hard, as has previously been shown for the random regular graphs measured in a product basis<sup>1</sup>. While this bears similarities to random circuit sampling<sup>2</sup>, by preparing random graph states, we can:

1. Generate highly entangled states.
2. Employ spacetime error detection<sup>3</sup> to post-select errored shots and achieve high fidelities, even at large circuit widths and depths
3. Verify fidelity efficiently with direct fidelity estimation<sup>4</sup>, which requires fewer assumptions about the noise and does not extrapolates from smaller circuits, as for cross-entropy benchmarking (XEB)

While we only measure the fidelity of the graph state, with Pauli twirling, the graph state and rotated state fidelities are equal up to first order in infidelity<sup>5</sup>. We also make use of the ability to toggle between stabilizer and non-stabilizer bases with rotational Z gates, which are implemented on IBM's hardware as noiseless framechanges<sup>6</sup>.

As tensor network contraction costs scale with the entanglement (Schmidt rank) of the graph state, and extended Clifford simulators scale with the non-stabilizerness (stabilizer extent), we claim that sampling is adversarial for exact or naïve classical simulations.

## Experimental Details and Results

We prepare a random graph state on a 70 X 70 circuit on IBM Boston. We equip the circuit with 27 ancilla qubits and perform spacetime error detection.

<p align="center">
  <img width="468" height="322" alt="Image" src="https://github.com/user-attachments/assets/52ca7570-b809-49aa-b528-40ff33ba2311" />

  <em>Figure 1. Physical layout for the 70 X 70 circuit (with 27 ancilla qubits) on IBM Boston, a Heron R3 device with heavy-hex architecture. The logical qubits (green) are arranged on a 1D chain with dangling ancilla qubits (red).</em>

</p>

Properties:

- **Graph state fidelity: 0.37** (bounded above 0.092 with 95% confidence)
- Data qubits: [45, 37, 25, 24, 23, 22, 21, 36, 41, 42, 43, 56, 63, 62, 61, 76, 81, 82, 83, 96, 103, 104, 105, 106, 107, 97, 87, 88, 89, 78, 69, 70, 71, 58, 51, 50, 49, 38, 29, 30, 31, 32, 33, 39, 53, 54, 55, 59, 75, 74, 73, 79, 93, 92, 91, 98, 111, 110, 109, 118, 129, 128, 127, 126, 125, 124, 123, 136, 143, 144]
- Ancillas: [44, 26, 16, 40, 64, 60, 80, 84, 102, 117, 108, 86, 90, 68, 72, 52, 48, 28, 18, 34, 94, 112, 130, 137, 122, 142, 145]
- Lowest known Schmidt rank (graph state): $2^{30}$
- Lowest known Stabilizer extent: $2^{23.97}$

To verify the fidelity of the graph state, we employ direct fidelity estimation and measure the expectation values of 80 random stabilizers.

<p align="center">
  <img width="1211" height="611" alt="Image" src="https://github.com/user-attachments/assets/1ad6d842-dfe5-4827-b8d0-228182b901fa" />

  <em>Figure 2. Fidelity and post-selection rate for the 70 X 70 circuit on IBM Boston. Using direct fidelity estimation, we draw 80 random stabilizers and measure their expectation values or fidelities. Pictured are the 66% and 95% confidence intervals. </em>

</p>

We do not directly measure the fidelity of the rotated graph state. This is not efficient in general and, for direct fidelity estimation, would require exponentially many samples. As stated above, we instead infer the rotated graph state fidelity to be equal to the graph state fidelity under the Pauli Twirling assumption.

## Classical Runtime Estimation

We estimate the runtime of two classical approaches, tensor networks (matrix product state) and extended stabilizer simulations.

For tensor networks, the entanglement is the most relevant challenge. To accurately represent a state, a bond dimension equivalent to the Schmidt rank is necessary. The contraction time is estimated for the 70 qubit circuit with increasing depth (max depth 24), with an estimated contraction time of $10^{25}$ seconds.

<p align="center">
  <img width="584" height="455" alt="Image" src="https://github.com/user-attachments/assets/e8c5d4f4-19fe-4ff2-90bf-19caa0c645b2" />

  <em>Figure 3. Quimb matrix product state (MPS) contraction times for the 70 X 70 circuit with increasing depth (maximum depth 24). Linear extrapolation from a logarithmic plot of the data (R<sup>2</sup> > 0.98) yields a predicted contraction time of 10<sup>25</sup> seconds. </em>

</p>

For extended stabilizer simulations, the complexity scales with the non-stabilizerness of the state. We quantify this with the stabilizer extent, which, for the face state basis we use to rotate the state, depends exponentially on the size of the state. Simulations track the time to draw one sample from increasingly large N X N circuits (maximum N=42). The estimated runtime is $10^{7}$ seconds.

<p align="center">
  <img width="570" height="476" alt="image" src="https://github.com/user-attachments/assets/b94a6fe8-a237-468a-ab67-14fbb9057279" />

  <em>Figure 4. Extended stabilizer simulation timing in Qiskit Aer for N X N circuits (maximum N=42) with face state preparation. Linear extrapolation from a logarithmic plot of the data (R<sup>2</sup> > 0.98) yields a predicted time of 10<sup>7</sup> seconds to sample once.</em>

</p>

## Classical Verification Task

The challenge for classical algorithms is to demonstrate strong simulation, either with the same fidelity in shorter time or with a higher fidelity in the same time as the quantum experiment. We propose three methods to verify that classical methods can adequately simulate random graph sampling:

1. Given that the rotated graph state uses O(N) non-Clifford gates, it approximates a k-design<sup>7</sup> and is close to Haar-random. Therefore, the linear cross entropy benchmarking (XEB) score is a reasonable proxy for fidelity, assuming the noise is weak and uncorrelated. We provide quantum samples below, for which one could show that a classical simulation produces an XEB score significantly larger than 0.092.
2. Alternatively, as in Zhao et. al's work on simulating random circuit sampling<sup>8</sup>, it also suffices to demonstrate that uncorrelated samples from the rotated graph state can be produced with XEB score significantly larger than 0.092.
3. For classical simulations (e.g. tensor networks) for which it is equally difficult to represent the graph state and rotated graph state, and for which conversion for the two can be done with small error, it would suffice to simulate the graph state, perform direct fidelity estimation, and demonstrate fidelity above 0.37.

Quantum Samples in the non-Clifford basis (1726 bitstrings after post-selection):
- [samples_nq70_depth70_checks27_basis_fstate.npy.zip](https://github.com/user-attachments/files/27220052/samples_nq70_depth70_checks27_basis_fstate.npy.zip)
- (Bitstrings use Qiskit's little-endian notation)

## References

A detailed writeup can be found in the [circuit instance description](https://github.com/quantum-advantage-tracker/quantum-advantage-tracker.github.io/tree/main/data/classically-verifiable-problems/circuit-models/random_graph_sampling).

1. [Ghosh, Hangleiter, Helsen. Random Regular Graph States are Complex at Almost Any Depth, PRX 2025](https://journals.aps.org/prxquantum/abstract/10.1103/52xz-3hpc)
2. [Arute et al. Quantum supremacy using a programmable superconducting processor, Nature 2019](https://www.nature.com/articles/s41586-019-1666-5)
3. [Martiel, Javadi-Abhari. Low-overhead error detection with spacetime codes, arXiv:2504.15725](https://arxiv.org/abs/2504.15725)
4. [Flammia, Liu. Direct Fidelity Estimation from Few Pauli measurements, PRA 2011](https://link.aps.org/doi/10.1103/PhysRevLett.106.230501)
5. [Merkel et al. When Clifford benchmarks are sufficient; estimating application performance with scalable proxy ](https://arxiv.org/pdf/2503.05943)[arXiv:2503.05943 ](https://arxiv.org/pdf/2503.05943)
6. [McKay et al. Efficient Z Gates for Quantum Computing, PRA 2018](https://link.aps.org/doi/10.1103/PhysRevA.96.022330)
7. [Leone et al. Non-Clifford Cost of Random Unitaries, arXiv:2505.10110](https://arxiv.org/pdf/2505.10110)
8. [Zhao et al. Leapfrogging Sycamore: Harnessing 1432 GPUs for 7× Faster Quantum Random Circuit Sampling, arXiv:2406.18889](https://arxiv.org/pdf/2406.18889)
### Authors

Ali Javadi-Abhari, Simon Martiel, Jay-U Chung, Alireza Seif, Soumik Ghosh

### Institutions

IBM, UChicago

### Quantum runtime (seconds)

1950

### Classical runtime (seconds)

10^7 (Qiskit Aer Extended Stabilizer)

10^25 (Quimb Matrix Product State)

### Compute resources (quantum)

IBM Heron R3

### Compute resources (classical)

Intel Xeon Gold 6258R 28 cores with 1 TB memory (Qiskit Aer), Apple M1 Max 10 cores with 32 GB memory (Quimb)

### Notes

The reported classical runtimes are extrapolated.

**Comment by n-e-sherman** [comment]
Hi @jchung2020, thank you for the submission. This is an interesting problem and you provide a quite thorough explanation of your results. You provide classical comparison using both MPS and extended stabilizer based simulations, as well as error-detection and post-selection of the quantum results. 

One minor comment is that I found a more thorough explanation with the circuits at the link below. It would be good to either include them here, or explicitly point to this link in the submission.
https://github.com/quantum-advantage-tracker/quantum-advantage-tracker.github.io/tree/main/data/classically-verifiable-problems/circuit-models/random_graph_sampling

I have a few questions around the specifics on the experiment and what precisely the task is that you are claiming quantum advantage for. If I understand your strategy correctly, you do the following (let $|G\rangle$ be the Clifford stabilizer state, and $|R\rangle$ the rotated state)
- You compute DFE for $|G\rangle$.
- You argue that the DFE for $|R\rangle$ is the same to first order in the infidelity due to noiseless virtual rotations.
- You generate 1726 samples from $|R\rangle$.
- You claim that these samples produce an XEB score of 0.37.  If I understand correctly, this comes from the fact that your mean DFE for $|G\rangle$ is 0.37, but is not computed directly using exact amplitudes of $|R\rangle$.

If I have this right, the directly measured quantity is DFE for $|G\rangle$, with the DFE and XEB for $|R\rangle$ inferred via framechange/twirling and anticoncentration arguments respectively. Please let me know if I've misunderstood. If correct, it would help to:
1. State the classical task more precisely, 
2. Flag which reported numbers are inferred vs. measured
3. Justify why the DFE equivalence of $|G\rangle$ and $|R\rangle$ is available on the quantum side but not the classical side, where classically $|G\rangle$ permits an efficient representation with fidelity 1. For example, one could argue that measuring the fidelity of $|R\rangle$ is still hard with quantum due to small expectation values and excessive shots needed, while for classical simulation, the difficulty is coming from a large stabilizer extent. So it would help to articulate why this is valid for quantum data but not classically.

**Comment by ajavadia** [comment]
Hi @n-e-sherman 

Thanks for your comments. Yes your understanding is correct. I'd just point out that our bound is not merely on the XEB of $|R\rangle$, but rather on its fidelity. Fidelity is a stronger measure than XEB, as they can diverge under certain distributions or certain noise. 

Nevertheless we believe obtaining a high XEB score through classical simulation is a good refutation of our quantum advantage claim. This is because the state ensemble that we generate is a random stabilizer state with O(n) non-Clifford gates, so the second moment is Porter-Thomas like. 

You are right that directly measuring the fidelity of $|R\rangle$ on the quantum hardware is difficult due to small expectation values. This is why we infer it from the fidelity of $|G\rangle$ which is efficient. But the task is generating samples from $|R\rangle$. Fidelity estimation is just a way for us to have confidence in our generated samples.

**Comment by n-e-sherman** [comment]
Hi @ajavadia 

Thanks for the clarification, that is helpful. If I understand correctly, the primary task is sampling from $|R\rangle$, and the DFE discussion is providing evidence that your samples are high quality. So the classical task would be something like sampling from $|R\rangle$ in a manner that is on-par or better than what is shown here, such as faster at the same quality, or a higher quality in the same time. To verify such a claim, you propose that a classical sampling strategy that can achieve a state fidelity significantly above 0.01, or one that achieves an XEB $\gtrsim$ 0.37, would be sufficient.

If that's correct, then I am happy to accept this submission after the following minor changes:
- Add the link to the circuits and more detailed writeup to the submission.
- Clarify the verification task to include some of these subtleties.

It seems like there is some gray area around precisely what would qualify as a successful classical solution. I think that is fine, but it would be good to discuss this a bit. In particular, your proposed tasks would be sufficient, but are not strictly necessary if by some other means one can demonstrate that they can efficiently generate high quality samples from $|R\rangle$ classically.

**Comment by ajavadia** [comment]
Yes. The attempt at classically solving the problem would be similar to previous attempts at simulating quantum supremacy / RCS experiments. For example: https://arxiv.org/abs/2406.18889

This involves generating samples classically that satisfy a high XEB (if you can simulate the circuit you can also calculate XEB). Or it could be by showing that the bitstring probabilities satisfy Porter-Thomas distribution.

@jchung2020 could you please update the text to reflect this discussion?

**Comment by Roland-Farrell** [comment]
Hi all,
I agree with Nick and think this is a great problem to add to the tracker once the comments get addressed.

**Comment by jchung2020** [comment]
Thank you for the feedback! I have made the proposed changes and added some concrete verification tasks. @ajavadia let me know if these look reasonable

**Comment by n-e-sherman** [comment]
Thanks for the updates @jchung2020, these look good to me.

**Comment by ajavadia** [comment]
looks good to me too thanks @jchung2020
